### ISSUE TITLE
Maybe `Frag: Send`?

### ISSUE THREAD

**Issue by jeehoonkang** [spacejam/rsdb#89]
I am trying to port `rsdb` to `crossbeam-epoch`: https://github.com/crossbeam-rs/crossbeam-epoch/

In doing so, I found out that `Stack` probably should require `T: Send`, because an item can be sent to elsewhere.  In turn, I guess `PageCache`'s `P` should be `Send` because of `inner: Radix<Stack<CacheEntry<P>>>`, and then `Frag` should be `Send`.  What do you think?

**Comment by spacejam** [comment]
@jeehoonkang ahh, interesting that I missed this, those should definitely be `Send`!

I'm looking forward to the port to `crossbeam-epoch`!

**Comment by spacejam** [comment]
Would it be more convenient for you if I implemented that right now? If not, feel free to add it to your port, and I'd gladly accept a PR for using `crossbeam-epoch`!

**Comment by jeehoonkang** [comment]
Honestly, I would like to ask if you could add `: Send`.  In fact, I tried to add `Send` on my own, but I currently couldn't understand the codebase and I am not sure what is the best way to do it: exactly where should I put `: Send`?

FYI, except for this problem, it seems `rsdb` is successfully compiled with `crossbeam-epoch`'s PR on garbage management: https://github.com/crossbeam-rs/crossbeam-epoch/pull/5

**Comment by spacejam** [comment]
I'll do that now. I think the best place to implement it at first will be on the requirement of the wrapped `T` in the stack and radix trees, as those are the underlying shared containers that threads share state with. This is sort of a specialized stack for this use case and I don't care about supporting single-threaded usage by conditionally allowing non-`Send` wrapped `T`'s.

This will be out in like 10 minutes, stay tuned!

**Comment by spacejam** [comment]
Also need `Sync` because multiple threads may be iterating over the same stacks to materialize views of pages simultaneously...

**Comment by spacejam** [comment]
waiting on https://github.com/spacejam/rsdb/pull/90 to pass tsan/lsan then merging and publishing version `0.8.0`.

I did not implement `Send` on `Frag` because it should have been auto-derived, due to consisting entirely of other types that are `Send`. @jeehoonkang please let me know if this addresses the issue or not! I'll post here again after publishing the crate.

**Comment by spacejam** [comment]
@jeehoonkang `0.8.0` published, let me know if that works for you!

**Comment by jeehoonkang** [comment]
Thanks!  I ported `rsdb` to `crossbeam-epoch`: https://github.com/jeehoonkang/rsdb/commit/dc533833fefd07ec8ab1e2b9b4e1075490c1b52e

`cargo test` worked, but I couldn't run lsan and tsan yet.

I needed to add `T: Sync` for several methods in `Stack`: https://github.com/jeehoonkang/rsdb/commit/dc533833fefd07ec8ab1e2b9b4e1075490c1b52e#diff-b9fd53db7195b303c4713ea2c461ed52R146  This is because they call `defer_drop`, which requires `T: Sync`.

Anyway I think now we can close this issue.
